packages/wallet/backend/tests/transaction/service.test.ts

```diff
@@ -0,0 +1,299 @@
+import { Bindings } from '@/app'
+import { env } from '@/config/env'
+import { Container } from '@/shared/container'
+import { createApp, TestApp } from '@/tests/app'
+import { Knex } from 'knex'
+import { truncateTables } from '@/tests/tables'
+import { Request, Response } from 'express'
+import {
+  createRequest,
+  createResponse,
+  MockRequest,
+  MockResponse
+} from 'node-mocks-http'
+import type { AuthService } from '@/auth/service'
+import { applyMiddleware } from '@/tests/utils'
+import { withSession } from '@/middleware/withSession'
+import {
+  mockedListAssets,
+  mockedTransactionInsertObjs,
+  mockLogInRequest
+} from '../mocks'
+import { TransactionService } from '@/transaction/service'
+import { Transaction } from '@/transaction/model'
+import { AccountService } from '@/account/service'
+import { PaymentPointerService } from '@/paymentPointer/service'
+import { faker } from '@faker-js/faker'
+import { User } from '@/user/model'
+import { createUser } from '@/tests/helpers'
+
+describe('Transaction Controller', (): void => {
+  let bindings: Container<Bindings>
+  let appContainer: TestApp
+  let knex: Knex
+  let authService: AuthService
+  let accountService: AccountService
+  let paymentPointerService: PaymentPointerService
+  let transactionService: TransactionService
+  let req: MockRequest<Request>
+  let res: MockResponse<Response>
+  let userId: string
+
+  const args = mockLogInRequest().body
+
+  // "dependency" = ALL foreign keys (paymentPointers, account, user...)
+  const prepareTransactionDependencies = async () => {
+    const accountServiceDepsMocked = {
+      rafiki: {
+        getAssetById: (id: unknown) =>
+          mockedListAssets.find((asset) => asset.id === id)
+      },
+      rapyd: {
+        issueVirtualAccount: () => ({
+          status: {
+            status: 'SUCCESS'
+          },
+          data: {
+            id: 'mocked'
+          }
+        }),
+        simulateBankTransferToWallet: () => ({
+          status: {
+            status: 'SUCCESS'
+          },
+          data: {
+            transactions: [
+              {
+                id: 'mocked'
+              }
+            ]
+          }
+        }),
+        getAccountsBalance: () => ({
+          data: [
+            {
+              currency: mockedListAssets[0].code,
+              balance: 777
+            }
+          ] as Partial<RapydAccountBalance>
+        })
+      }
+    }
+    const ppServiceDepsMocked = {
+      accountService: await bindings.resolve('accountService'),
+      env: await bindings.resolve('env'),
+      rafikiClient: {
+        createRafikiPaymentPointer: () => ({
+          id: faker.string.uuid(),
+          url: faker.string.alpha(10)
+        })
+      }
+    }
+    Reflect.set(accountService, 'deps', accountServiceDepsMocked)
+    Reflect.set(paymentPointerService, 'deps', ppServiceDepsMocked)
+
+    const account = await accountService.createAccount({
```

Comment:
I suggest to insert the fake account directly in db
in transactions test there is no value in mocking all the things that account service needs to create an account